### PR TITLE
Refactored events to also handle on_init, on_load and on_configuration changed

### PR DESCRIPTION
**Pull request by mojo2012**
Now we can use on_init, on_load and on_confugration_changed the same way:
* Event.register(core_events.init, function() 
* Event.register(core_events.load, function() 
* Event.register(core_events.configuration_changed, function()

### CODE REVIEW

File: stdlib/event/event.lua [hunk 1]
```diff
@@ -4,8 +4,25 @@
 require 'stdlib/core'
 require 'stdlib/game'
 
+core_events = { init = -1, load = -2, configuration_changed= -3 }
+
 Event = {_registry = {}}
 
+script.on_init(function()
```
Comment (Afforess) [inline stdlib/event/event.lua]: I like the idea of adding the core events to the registry, but this presents a problem as currently implemented. Say a mod is currently using the Event registry, and then requires 'stdlib/event/event', won't these functions override what a mod might already have set as the handler?

I think we need to add the script.on_init, on_load, and on_configuration_changed in the Event.register call, when a core event is registered, so we know we are setting the handler for a mod that explicitly requested a core handler.

File: stdlib/event/event.lua [hunk 2]
```diff
@@ -4,8 +4,25 @@
 require 'stdlib/core'
 require 'stdlib/game'
 
+core_events = { init = -1, load = -2, configuration_changed= -3 }
```
Comment (Afforess) [inline stdlib/event/event.lua]: I'd move core_events to the Event object on line 9 below, so we aren't polluting the space with too many globals. Maybe call it Event.core_events.init or even Event.core.init... One of the two anyway.

**Comment by mojo2012** [comment]
As I experienced it, mods would override the Event scripts init handler, as the last always wins - and normally you have yor includes at the top of the file.

But still it might be good to implement it as you suggested. I'm just to unfamiliar with lua script to implement it in a fine way. Additionally I was not clear on how the objects passed look like exactly (the problem of dynamic languages plus no debug more :( ).

**Comment by Afforess** [comment]
@mojo2012 I made some tweaks to your PR, see what you think: https://github.com/Afforess/Factorio-Stdlib/pull/16

I think it solves the issues I brought up, and it added some tests for the new core events.

**Pull request merged**